src/main/java/org/truffleruby/core/regexp/MatchDataNodes.java

```diff
@@ -167,17 +167,56 @@ protected Object create(Object regexp, Object string, int start, int end) {
     @Primitive(name = "matchdata_create")
     public abstract static class MatchDataCreateNode extends PrimitiveArrayArgumentsNode {
 
+        public static MatchDataCreateNode create() {
+            return MatchDataNodesFactory.MatchDataCreateNodeFactory.create(null);
+        }
+
+        public abstract Object executeMatchDataCreate(RubyRegexp regexp, Object string, Object starts, Object ends);
+
         @Specialization
-        protected Object create(Object regexp, Object string, RubyArray starts, RubyArray ends,
+        protected Object matchDataCreate(RubyRegexp regexp, Object string, RubyArray starts, RubyArray ends,
```

Comment:
This is for the other caller in strscan.rb and specs, I see.
Looks like we can use `Primitive.matchdata_create_single_group` for those and then drop this specialization.